src/components/common/portal/Portal.tsx

```diff
@@ -0,0 +1,20 @@
+import { ReactNode } from "react";
+import { createPortal } from "react-dom";
+
+interface PortalProps {
+  children: ReactNode;
+  container: string;
+  mounted: boolean;
+}
+
+function Portal({ children, container, mounted }: PortalProps) {
+  if (mounted) {
+    const portal = document.querySelector(container);
+
+    return portal ? createPortal(children, portal) : null;
+  }
+
+  return null;
+}
```

Comment:
이렇게 Portal 사용하는거 첨보는데 깔끔하고 좋네용 👍

layout에 선언되는 modal-root같은 컨테이너?의 아이디로 선언하는 형태는 깨지지 않을텐데 #modal-root 처럼 prop에 #을 붙이면서 querySelector로 받아와야하는 이유가 있을까여?

```suggestion
function Portal({ children, containerId, isMounted }: PortalProps) {
  if (isMounted) {
    const portal = document.getElementBtId(containerId);

    return portal ? createPortal(children, portal) : null;
  }

  return null;
}
```

Reply:
둘 중 어느 방법이어도 상관이 없을 것 같았는데 ! 
혹시 특별한 차이점이 있을까요 ?! 
혹은 '#'을 붙이는 부분이 어색한 것일까요 ?!
우선은 저도 getElementBtId 부분이 더 익숙한 것 같아서 수정하도록 하겠습니다!

Reply:
지금 당장은 그럴 일이 드물겠지만 고도화된다면 실수 방지 및 트래킹 위해서 상수화도 고려하면 좋을텐데(비단 해당 부분이 아니더라도) id로 사용하고 있다면 그대로 같은 id 값으로 로직이 진행되는 편이 자연스럽다는 생각이에요.
portal에 `.something` 처럼 클래스는 쓸 일이 없을거고 id가 유일할텐데 이런 점에서 querySelector는 불필요한 추상화만 더해지는 것 같아영